File: exercises/queue/queue.py
class Queue:
    def __init__(self):
        self.data = []

    def add(self, record):
        self.data.insert(0, record)

    def remove(self):
        return self.data.pop()

    def peek(self):
        return self.data[-1]


q = Queue()

q.add(1)
q.add('a')
print(q.remove())
print(q.peek())

File: exercises/weave/queue.js
// --- Directions
// Implement a 'peek' method in this Queue class.
// Peek should return the last element (the next
// one to be returned) from the queue *without*
// removing it.

class Queue {
  #data;
  constructor() {
    this.#data = [];
  }

  add(record) {
    this.#data.unshift(record);
  }

  remove() {
    return this.#data.pop();
  }

  // to see what the next value would/check if there is one
  peek() {
    return this.#data.slice(-1)[0];
  }
}

module.exports = Queue;
